api/middlewares/auth.js

```diff
@@ -0,0 +1,25 @@
+const config = require('config');
+
+/**
+ * check the user's api key
+ * @param {Object} req - HTTP request
+ * @param {Object} res - HTTP response
+ * @param {function} next - do the following
+ * @returns {Object|function} res or next
+ */
+const checkAuth = (req, res, next) => {
+  let apikeyusers = config.get('apikeyusers');
+  try {
+    apikeyusers = JSON.parse(apikeyusers);
+  } catch (err) {
+    return res.status(500).json({ message: 'Internal server error in JSON.parse(apikeyusers)' });
+  }
+  if (!apikeyusers.includes(req?.headers?.api_key)) {
```

Comment:
Express has a shorthand to get header values.
You may also use a `Set` or a `Map` to speedup searches (`includes` actually loops over array elements). However, I'm not sure about the actual performance impact since we don't have much keys for now.

```suggestion
  if (!apikeyusers.includes(req.get('api_key')) {
```